spec/models/miq_queue_spec.rb

```diff
@@ -6,13 +6,132 @@
       _, @miq_server, @zone = EvmSpecHelper.create_guid_miq_server_zone
     end
 
-    it "works with deliver_on" do
+    it "requires class_name" do
+      msg = MiqQueue.new(:class_name => nil)
+      status, message, result = msg.deliver
+      expect(status).to eq(MiqQueue::STATUS_ERROR)
+      expect(message).to eq("class_name cannot be nil")
+      expect(result).to be_nil
+    end
+
+    it "requires valid class_name" do
+      msg = MiqQueue.new(:class_name => "NotARealClazz")
+      status, message, result = msg.deliver
+      expect(status).to eq(MiqQueue::STATUS_ERROR)
+      expect(message).to eq("uninitialized constant NotARealClazz")
+      expect(result).to be_nil
+    end
+
+    it "uses class_name without instance_id" do
+      expect(MiqServer).to receive(:my_zone).and_return("MY ZONE")
+      msg = MiqQueue.new(:class_name => "MiqServer", :method_name => "my_zone")
+
+      status, message, result = msg.deliver
+      expect(status).to eq(MiqQueue::STATUS_OK)
+      expect(message).to eq("Message delivered successfully")
+      expect(result).to eq("MY ZONE")
+    end
+
+    it "uses object with instance_id" do
+      expect(MiqServer).to receive(:find).with(@miq_server.id).and_return(@miq_server)
+      expect(@miq_server).to receive(:my_zone).and_return("MY ZONE")
+      msg = MiqQueue.new(:class_name => "MiqServer", :instance_id => @miq_server.id, :method_name => "my_zone")
+
+      status, message, result = msg.deliver
+      expect(status).to eq(MiqQueue::STATUS_OK)
+      expect(message).to eq("Message delivered successfully")
+      expect(result).to eq("MY ZONE")
+    end
+
+    it "handles record not found" do
+      msg = MiqQueue.new(:class_name => "MiqServer", :instance_id => @miq_server.id + 100, :method_name => "my_zone")
```

Comment:
hardcoded id increments such as `+100` scare me.  I'm not sure how else to test this though so ☹️

Reply:
yea. if you have another idea of forcing a record not found. I could stub it... (which you and I like to avoid)

Reply:
maybe order by id, pluck them and +1 on the largest value?